metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/ESSearchDAO.java

```diff
@@ -262,7 +269,27 @@ public ScrollResult scroll(@Nonnull List<String> entities, @Nonnull String input
 
     scrollRequestTimer.stop();
     // Step 2: execute the query and extract results, validated against document model as well
-    return executeAndExtract(entitySpecs, searchRequest, postFilters, scrollId, keepAlive, size);
+    return executeAndExtract(entitySpecs, searchRequest, transformedFilters, scrollId, keepAlive, size);
+  }
+
+  /**
+   * Allows filtering on entities which are stored as different indices under the hood by transforming the tag
+   * _entityType to _index and updating the type to the index name.
+   * @param filter The filter to parse and transform if needed
+   * @param indexConvention The index convention used to generate the index name for an entity
+   * @return A filter, with the changes if necessary
+   */
+  static Filter transformFilterForEntities(Filter filter, @Nonnull IndexConvention indexConvention) {
+    if (filter != null && filter.getOr() != null) {
```

Comment:
Some of this logic might be able to be reduced with usage of QueryUtils, filters are kind of messy to construct so we have some utility methods around them :)

Reply:
Took a look but I don't think these will work for this case :( but good to know about this!